ablySpec/Auth.swift

```diff
@@ -529,5 +529,148 @@ class Auth : QuickSpec {
+
+                let authOptions = ARTAuthOptions(key: "key:secret")
+
+                auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                    expect(tokenRequest?.keyName).to(equal("key"))
+                })
+            }
+
+            // RSA9c
+            it("should generate a unique 16+ character nonce if none is provided") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                    expect(tokenRequest?.nonce.characters).to(haveCount(16))
+                })
+            }
+
+            // RSA9d
+            context("should generate a timestamp") {
+
+                it("from current time if not provided") {
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                    rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                        expect(tokenRequest?.timestamp).to(beCloseTo(NSDate(), within: 1.0))
+                    })
+                }
+
+                it("will retrieve the server time if queryTime is true") {
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                    var serverTime: NSDate?
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.time({ date, error in
+                            serverTime = date
+                            done()
+                        })
+                    }
+
+                    expect(serverTime).toNot(beNil(), description: "Server time is nil")
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.queryTime = true
+
+                    rest.auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                        expect(tokenRequest?.timestamp).to(beCloseTo(serverTime!, within: 6.0))
+                    })
+                }
+            }
+
+            // RSA9e
+            it("should TTL be optional and specified in milliseconds") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams()
+                let expectedTtl = 6.0
+                tokenParams.ttl = expectedTtl
+
+                rest.auth.createTokenRequest(tokenParams, options: nil, callback: { tokenRequest, error in
+                    //In Seconds because TTL property is a NSTimeInterval but further it does the conversion to miliseconds
+                    expect(tokenRequest?.ttl).to(equal(expectedTtl))
+                })
+            }
+
+            // RSA9f
+            it("should provide capability has json text") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams()
+                let expectedCapability = "{ - }" //Invalid
```

Comment:
If this is passing, then it indicates something is wrong with your implementation because that would fail.

```
$ curl -X POST "https://rest.ably.io/keys/xVLyHw.OpNddg/requestToken" \
>  -u "xVLyHw.OpNddg:tFG_5gBPYR9e_LtA" \
>  -H "Content-Type: application/json" \
>  --data '{ "keyName": "xVLyHw.OpNddg", "timestamp": 1449091700854, "capability": "{ - }" }'
{
	"error": {
		"statusCode": 400,
		"code": 40000,
		"message": "Malformed capability",
		"serverId": "frontend.2c91.production-eu-west-1-A.i-b1cf411b"
	}
}
```

So please look into why it is passing with an invalid capability, and make sure you add a test to ensure we don't get a regression please.  Once fixed, then we can look at merging this in